pkg/pgmodel/metrics/ingest.go

```diff
@@ -0,0 +1,210 @@
+// This file and its contents are licensed under the Apache License 2.0.
+// Please see the included NOTICE for copyright information and
+// LICENSE for a copy of the license.
+
+package metrics
+
+import (
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/timescale/promscale/pkg/util"
+)
+
+// Keep these const values in pgmodel/metrics to avoid cyclic import since 'trace' may also need this.
+const (
+	MetricBatcherChannelCap = 1000
+	// FlushSize is the maximum number of insertDataRequests that should be buffered before the
+	// insertHandler flushes to the next layer. We don't want too many as this
+	// increases the number of lost writes if the connector dies. This number
+	// was chosen arbitrarily.
+	FlushSize           = 2000
+	MaxInsertStmtPerTxn = 100
+)
+
+var (
+	// MaxSentTs is the max timestamp sent to the database.
+	MaxSentTs          = int64(0)
+	IngestorDuplicates = prometheus.NewCounterVec(
+		prometheus.CounterOpts{
+			Namespace: util.PromNamespace,
+			Subsystem: "ingest",
+			Name:      "duplicates_total",
+			Help:      "Total number of processed samples/write_requests_to_db/metrics which where duplicates.",
+		}, []string{"type", "kind"},
+	)
+	IngestorDecompressCalls = prometheus.NewCounterVec(
+		prometheus.CounterOpts{
+			Namespace: util.PromNamespace,
+			Subsystem: "ingest",
+			Name:      "decompress_calls_total",
+			Help:      "Total number of calls to decompress_chunks_after.",
+		}, []string{"type", "kind"},
+	)
+	IngestorDecompressEarliest = prometheus.NewGaugeVec(
+		prometheus.GaugeOpts{
+			Namespace: util.PromNamespace,
+			Subsystem: "ingest",
+			Name:      "decompress_min_unix_time",
+			Help:      "Earliest decompression time in unix.",
+		}, []string{"type", "kind", "table"})
+	HAClusterLeaderDetails = prometheus.NewGaugeVec(
+		prometheus.GaugeOpts{
+			Namespace: util.PromNamespace,
+			Subsystem: "ingest",
+			Name:      "ha_cluster_leader_info",
+			Help:      "Info on HA clusters and respective leaders.",
+		},
+		[]string{"cluster", "replica"})
+	NumOfHAClusterLeaderChanges = prometheus.NewCounterVec(
+		prometheus.CounterOpts{
+			Namespace: util.PromNamespace,
+			Subsystem: "ingest",
+			Name:      "ha_cluster_leader_changes_total",
+			Help:      "Total number of times leader changed per cluster.",
+		},
+		[]string{"cluster"})
```

Comment:
I don't think those metrics play part in only `ingest` component of promscale and as such I don't think they should be in `ingest` Subsystem. IMHO they should live in `ha` subsystem.